Sport1Player/Sport1PlayerAdapter.m

```diff
@@ -318,24 +328,95 @@ -(void)shouldPresentPin {
         }
     }];
 }
-#pragma mark - Livestream Token
--(NSObject <ZPPlayable>*)amendIfLivestream:(NSObject <ZPPlayable>*)current {
-    if (current.isLive) {
-        NSString *streamToken = [[[ZAAppConnector sharedInstance] storageDelegate] sessionStorageValueFor:kTokenName
-                                                                                                namespace:kNameSpace];
-        
-        if (streamToken == nil) {return current;}
-        
-        NSString *url = current.contentVideoURLPath;
-        NSString *amendedURL = [[NSString alloc] initWithFormat:@"%@?access_token=%@", url, streamToken];
-        Sport1StreamPlayable *amended = [[Sport1StreamPlayable alloc] initWithOriginal:current
-                                                                         andAmendedURL:amendedURL];
-        return amended;
-    } else {return current;}
-}
+
 #pragma mark - Handlers
+
 -(void)applicationWillEnterForegroundNotificationHandler {
     [self shouldPresentPin];
 }
 
+#pragma mark - Livestream Token
+
+- (void)amendIfLivestreamModified:(NSObject <ZPPlayable>*)current
+                         callback:(void (^)(NSObject <ZPPlayable>* amended))completion {
+
+    if (!current.isLive) {
+        completion(current);
+        return;
+    }
+
+    [Sport1PlayerAdapter getStreamTokenFromAPI:^(BOOL success, NSString *streamToken) {
+        if (!success || !streamToken) {
+            completion(current);
+            return;
+        }
```

Comment:
If the API returns a 401 or 403 we should launch the login again.

Reply:
Yes, i'm using exactly this object.
My test scenarion:
- open eSport1 app
- go to second Tab (Live)
- catch break point in method amendIfLivestreamModified

Reply:
We are having problems in that tab at the moment. Please select the first video from the home screen. It is also a livestream.

Reply:
Ok, works fine with video from the first tab.